.github/workflows/cypress.yml

```
@@ -0,0 +1,24 @@
+name: Cypress
+
+on:
+    push:
+        branches:
+        - main
+    pull_request:
+        branches:
+        - main
+
+jobs:
+    build:
+        runs-on: ubuntu-latest
+        defaults:
+            run:
+              working-directory: ./publish-frontend
+        steps:
+        - uses: actions/checkout@v2
+        - name: Install dependencies
+          run: npm install
+        - name: Start server
+          run: npm run dev &
+        - name: Run tests
+          run: npm run cy:run
```

```suggestion
        - name: Cypress run
          uses: cypress-io/github-action@v6
          with:
            start: npm run dev
            wait-on: 'http://localhost:3000'
```
Their action handles a bunch of things (waiting for servers, caching, installing deps), so we might as well make use of it.